cranelift/2025/cranelift-04-02.md

```diff
@@ -18,5 +18,53 @@
 
 ### Attendees
 
+- fitzgen
+- alexcrichton
+- abrown
+- cfallin
+- Rahul Chaphalkar
+- uweigand
+
 ### Notes
 
+- loading return values with separate VCode instructions
+  - cfallin: issue came up with exception support: we can't have loads that are
+    logically part of the call as separate trailing VCode instructions, because
+    the call is a terminator now. Tried splitting edges, way too
+    ugly/unworkable; we need to do the loads as part of one pseudoinst.
+    - filed https://github.com/bytecodealliance/wasmtime/pull/10488 with all
+      the gory details of all the approaches tried
+  - this is now actually basically resolved by
+    https://github.com/bytecodealliance/wasmtime/pull/10502, and exceptions are
+    unblocked
+  - remaining issues: do we simplify ABI code by supporting only register
+    returns, and pushing stack returns to "userspace" (Cranelift
+    user/embedder)? And related: issues with struct-ret
+    - probably not for now -- if it works, it's actually nice to have the
+      uniform support and not have the frontend have to think too hard about
+      ABI-specific issues
+    - struct-ret issues are a separate question; follow up with `cg_clif` to
+      see what we can do, e.g. provide the primitive to put the pointer in the
+      right register (not necessarily one of the normal arg/return registers)
```

Comment:
We already have that primitive (`ArgumentPurpose::StructReturn`) and in fact cg_clif uses it for the C ABI. It doesn't use it for the Rust ABI however, but that probably needs some rustc changes.